app/services/gias_csv_importer.rb

```diff
@@ -60,13 +134,40 @@ def call
     end
     Rails.logger.silence do
       School.upsert_all(records, unique_by: :urn)
+      associate_schools_to_regions
     end
 
     Rails.logger.info "Done!"
   end
 
   private
 
+  def associate_schools_to_regions
+    Rails.logger.debug "Associating schools to regions... "
+
+    School.in_batches(of: 1000).each_record do |school|
+      region = determine_region(school.district_admin_code)
+      success = School.where(id: school.id).update_all(region_id: region.id)
+
+      unless success
+        Rails.logger.info "Failed to update region for school with ID #{school.id}"
+      end
+    end
+  end
+
+  def determine_region(district_admin_code)
+    case district_admin_code
+    when *INNER_LONDON_AREAS
+      Region.find_by(name: "Inner London")
```

Comment:
These should be Region.find_or_create_by!(name: "Outer London", ...)
Given that there is a possibility at this point that Regions might not exist.
Do we need a rake task in either this PR or https://github.com/DFE-Digital/itt-mentor-services/pull/201 to create Regions if they do not exist.